torch/csrc/jit/codegen/cuda/lower_validation.cpp

```diff
@@ -11,6 +11,50 @@ namespace jit {
 namespace fuser {
 namespace cuda {
 
+namespace {
+class ValidateParallelType : public IterVisitor {
```

Comment:
What is this class meant to do? Looks like it is not just validating but also modifying parallelization.

Reply:
This class just makes sure that all the outputs of `WelfordOp` have the same parallelization and automatically converts the un-parallelized outputs if only one of them is parallelize. This way API side does not need to specify all the axes three times. It needs to be done after all the `computeAt` replays finish and before the actual lowering starts. It's also too small a change to be a separate pass. Actually it is not the only place that mutates the Fusion IR in lower_validation pass. Maybe in a followup PR I can group these together and name them `LowerPreparation` or similar.

Reply:
Thanks for the explanation. Please leave it as a code comment.